processing/src/main/java/org/apache/druid/java/util/metrics/OshiSysMonitor.java

```diff
@@ -72,22 +76,34 @@ public class OshiSysMonitor extends FeedDefiningMonitor
   private final TcpStats tcpStats;
 
   private final Map<String, String[]> dimensions;
+  private final Set<String> skipEmitting;
+  private final Map<String, Consumer<ServiceEmitter>> monitoringFunctions = ImmutableMap.of(
+      "mem", this::monitorMemStats,
+      "swap", this::monitorSwapStats,
+      "fs", this::monitorFsStats,
+      "disk", this::monitorDiskStats,
+      "net", this::monitorNetStats,
+      "cpu", this::monitorCpuStats,
+      "sys", this::monitorSysStats,
+      "tcp", this::monitorTcpStats
+  );
 
   public OshiSysMonitor()
```

Comment:
This class seems to have too many constructors (we seem to be adding one more in this PR), only one of which is needed in production code (used in `MetricsModule`).

We should have at most one more constructor for tests, so in total 2 constructors.